src/Microsoft.ServiceFabric.Actors/Diagnostics/EventSourceDiagnosticEvents.cs

```diff
@@ -0,0 +1,189 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Fabric;
+using Microsoft.ServiceFabric.Actors.Runtime;
+using Microsoft.ServiceFabric.Diagnostics;
+using Microsoft.ServiceFabric.Services.Remoting;
+
+namespace Microsoft.ServiceFabric.Actors.Diagnostics
+{
+    internal class EventSourceDiagnosticEvents : IDiagnosticEvents
+    {
+        readonly ServiceContext serviceContext;
+        readonly IClock clock;
+        readonly string actorType;
+        readonly ActorFrameworkEventSource eventSource;
+        readonly Dictionary<long, ActorMethodInfo> actorMethodInfo;
+
+        public EventSourceDiagnosticEvents(ActorFrameworkEventSource eventSource, IClock clock, ServiceContext serviceContext, ActorMethodFriendlyNameBuilder nameBuilder, ActorTypeInformation typeInfo)
+        {
+            this.eventSource = eventSource ?? throw new ArgumentNullException(nameof(eventSource));
+            this.serviceContext = serviceContext;
+            this.clock = clock ?? throw new ArgumentNullException(nameof(clock));
+            _ = nameBuilder ?? throw new ArgumentNullException(nameof(nameBuilder));
+            _ = typeInfo ?? throw new ArgumentNullException(nameof(typeInfo));
+            this.actorType = typeInfo.ImplementationType.ToString();
+            this.actorMethodInfo = InitializeActorMethodInfo(nameBuilder, typeInfo);
+        }
+
+        private Dictionary<long, ActorMethodInfo> InitializeActorMethodInfo(ActorMethodFriendlyNameBuilder nameBuilder, ActorTypeInformation typeInfo)
```

Comment:
Does InitializeActorMethodInfo initialize this.actorMethodInfo or only builds it? Can this method be static?